test/test_serialize_load_commands.rb

```diff
@@ -0,0 +1,323 @@
+    filenames = SINGLE_ARCHES.map { |a| fixture(a, "hello.bin") }
+
+    filenames.each do |filename|
+      file = MachO::MachOFile.new(filename)
+      lc = file[:LC_UUID].first
+      blob = lc.view.raw_data[lc.view.offset, lc.cmdsize]
+
+      assert_equal blob, lc.serialize
+    end
+  end
+
+  def test_serialize_rpath
+    filenames = SINGLE_ARCHES.map { |a| fixture(a, "hello.bin") }
+
+    filenames.each do |filename|
+      file = MachO::MachOFile.new(filename)
+      lc = file[:LC_RPATH].first
+      cmd_round = file.magic32? ? 4 : 8
+      lc2 = MachO::LoadCommand.create(:LC_RPATH, cmd_round, lc.path.to_s)
+      blob = lc.view.raw_data[lc.view.offset, lc.cmdsize]
+
+      assert_equal blob, lc.serialize
+      assert_equal blob, lc2.serialize(lc.view.endianness)
+    end
+  end
+
+  def test_serialize_code_signature
+    pass
+  end
+
+  def test_serialize_segment_split_info
+    pass
+  end
+
+  def test_serialize_reexport_dylib
+    pass
+  end
+
+  def test_serialize_lazy_load_dylib
+    pass
+  end
+
+  def test_serialize_encryption_info
+    pass
+  end
+
+  def test_serialize_dyld_info
+    pass
+  end
+
+  def test_serialize_dyld_info_only
+    # ppc does not have a LC_DYLD_INFO_ONLY command
+    filenames = [:i386, :x86_64].map { |a| fixture(a, "hello.bin") }
+
+    filenames.each do |filename|
+      file = MachO::MachOFile.new(filename)
+      lc = file[:LC_DYLD_INFO_ONLY].first
+      blob = lc.view.raw_data[lc.view.offset, lc.cmdsize]
+
+      assert_equal blob, lc.serialize
+    end
+  end
+
+  def test_serialize_load_upward_dylib
+    pass
+  end
+
+  def test_serialize_version_min_macosx
+    # ppc does not have a LC_VERSION_MIN_MACOSX command
+    filenames = [:i386, :x86_64].map { |a| fixture(a, "hello.bin") }
+
+    filenames.each do |filename|
+      file = MachO::MachOFile.new(filename)
+      lc = file[:LC_VERSION_MIN_MACOSX].first
+      blob = lc.view.raw_data[lc.view.offset, lc.cmdsize]
+
+      assert_equal blob, lc.serialize
+    end
+  end
+
+  def test_serialize_version_min_iphoneos
+    pass
+  end
+
+  def test_serialize_function_starts
+    # ppc does not have a LC_FUNCTION_STARTS command
+    filenames = [:i386, :x86_64].map { |a| fixture(a, "hello.bin") }
+
+    filenames.each do |filename|
+      file = MachO::MachOFile.new(filename)
+      lc = file[:LC_FUNCTION_STARTS].first
+      blob = lc.view.raw_data[lc.view.offset, lc.cmdsize]
+
+      assert_equal blob, lc.serialize
+    end
+  end
+
+  def test_serialize_dyld_environment
+    pass
+  end
+
+  def test_serialize_main
+    pass
+  end
+
+  def test_serialize_data_in_code
+    pass
+  end
+
+  def test_serialize_source_version
+    # ppc does not have a LC_SOURCE_VERSION command
+    filenames = [:i386, :x86_64].map { |a| fixture(a, "hello.bin") }
+
+    filenames.each do |filename|
+      file = MachO::MachOFile.new(filename)
+      lc = file[:LC_SOURCE_VERSION].first
+      blob = lc.view.raw_data[lc.view.offset, lc.cmdsize]
```

Comment:
I agree, but having it be called `raw_data` might be confusing in light of `view.raw_data`. It's also a little confusing in terms of the presence of `serialize`, although not as much so.

Perhaps `raw_lc` (or just `raw`)?